Packed insertion method (#2033)
chaos-polymtl/lethe: doc/source/parameters/dem/insertion_info.rst

.. warning::
    The critical Rayleigh time step is computed from the parameters in the ``particle type`` subsections, not the ``insertion info`` subsection. It is the user's responsibility to fill the ``particle type`` subsections correctly according to the diameter values stored in the insertion input file, otherwise Rayleigh time percentage displayed at the start of every DEM simulation may not be accurate.

--------------------
Packed
--------------------

> The ``packed`` insertion method inserts particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions **as is** for the initial configuration of a DEM simulation would result in an unstable simulation. To solve this problem, overlapping particles are shifted apart, relative to their normal contact vector, until all particles are no longer touching other particles or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motions are not solved.

Reviewer: ```suggestion
The ``packed`` insertion method inserts particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions **as is** for the initial configuration of a DEM simulation would result in an unstable simulation. To solve this problem, overlapping particles are shifted apart, relative to their normal contact vector, until all particles are no longer touching other particles or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motion are not solved.
```

> The ``packed`` insertion method insert particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions as is for the initial configuration of a DEM simulation would result in a unstable simulation. To solve this problem, overlapping particles are shift apart, relative to their normal contact vector, until all particles are no longer touching each other or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motion on every particles are not solved.

Reviewer 2: ```suggestion
The ``packed`` insertion method insert particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions as is for the initial configuration of a DEM simulation would result in a unstable simulation. To solve this problem, overlapping particles are shifted apart, relative to their normal contact vector, until all particles are no longer touching each other or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motion on every particles are not solved.
```

> The ``packed`` insertion method insert particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions as is for the initial configuration of a DEM simulation would result in a unstable simulation. To solve this problem, overlapping particles are shift apart, relative to their normal contact vector, until all particles are no longer touching each other or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motion on every particles are not solved.

Reviewer: ```suggestion
The ``packed`` insertion method inserts particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions as is for the initial configuration of a DEM simulation would result in an unstable simulation. To solve this problem, overlapping particles are shifted apart, relative to their normal contact vector, until all particles are no longer touching each other or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motion on every particle are not solved.
```

> The ``packed`` insertion method inserts particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions as is for the initial configuration of a DEM simulation would result in an unstable simulation. To solve this problem, overlapping particles are shifted apart, relative to their normal contact vector, until all particles are no longer touching each other or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motion on every particle are not solved.

Reviewer 2: ```suggestion
The ``packed`` insertion method inserts particles in an insertion box in an unstructured manner without guaranteeing that the particles are not overlapping on insertion. Using these particle positions **as is** for the initial configuration of a DEM simulation would result in an unstable simulation. To solve this problem, overlapping particles are shifted apart, relative to their normal contact vector, until all particles are no longer touching other particles or walls. This particle shifting is performed using pseudo time-stepping, where gravity is not considered and where the equations of motions are not solved.
```

.. note::

>     Currently, when using this insertion method, the simulation ends once no contacts is detected, thus the particle positions needs to be extracted in an insertion file and the simulation must be relaunched with the ``file`` insertion method.

Reviewer: ```suggestion
    Currently, when using this insertion method, the simulation ends once no contact is detected, thus the particle positions need to be extracted in an insertion file and the simulation must be relaunched with the ``file`` insertion method.
```

.. note::

>     Due to the use of pseudo time-stepping and the functioning of this insertion method, if multiple particle types are used, every particles types will be inserted on the first insertion time-step.

Reviewer: ```suggestion
    Due to the use of pseudo time-stepping and the functioning of this insertion method, if multiple particle types are used, every particle type will be inserted on the first insertion time-step.
```

.. warning::

>     The ``packed`` insertion method is experimental. It does not guarantee that the particles will reach a non-overlapping configuration in a reasonable amount of time, especially when total inserted particle volume over the total triangulation volume is over 50%.

Reviewer: ```suggestion
    The ``packed`` insertion method is experimental. It does not guarantee that the particles will reach a non-overlapping configuration in a reasonable amount of time, especially when the total inserted particle volume over the total triangulation volume is over 50%.
```

>     The ``packed`` insertion method is experimental. It does not guarantee that the particles will reach a non-overlapping configuration is a reasonable amount of time, especially when total inserted particle volume over the total triangulation volume is over 50%.

Reviewer: ```suggestion
    The ``packed`` insertion method is experimental. It does not guarantee that the particles will reach a non-overlapping configuration in a reasonable amount of time, especially when total inserted particle volume over the total triangulation volume is over 50%.
```
